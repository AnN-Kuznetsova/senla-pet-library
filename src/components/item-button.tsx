import * as React from "react";
import {Button} from "@mui/material";


interface PropsType {
  textValue?: string,
  className?: string,
  onClick?: () => void,
}


export const ItemButton: React.FC<PropsType> = (props: PropsType) => {
  const {
    textValue,
    className,
    onClick,
  } = props;

  return (
    <Button
      variant="outlined"
      className={`item-button ${className}`}
      color="secondary"
      onClick={onClick}
    >
      {textValue}
      <span className="item-button__inner"></span>
    </Button>
  );
};
