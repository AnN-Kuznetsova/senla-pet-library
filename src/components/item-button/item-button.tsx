import * as React from "react";
import {Button} from "@mui/material";

import {ThemeMode} from "../../const";


export enum ItemButtonMode {
  CLOSE = `item-button--close`,
  DELETE = `item-button--delete`,
  MORE = `item-button--more`,
  ON_RIGHT = `item-button--on-right`,
  WARNING = `item-button--warning`,
}


interface PropsType {
  textValue?: string,
  className?: ItemButtonMode | `${ItemButtonMode} ${ItemButtonMode}`,
  isDisabled?: boolean,
  onClick: () => void,
}


export const ItemButton: React.FC<PropsType> = (props: PropsType) => {
  const {
    textValue,
    className,
    isDisabled,
    onClick,
  } = props;

  let buttonColor = ThemeMode.INFO;

  if (className) {
    switch (true) {
      case className.includes(ItemButtonMode.CLOSE):
      case className.includes(ItemButtonMode.DELETE):
        buttonColor = ThemeMode.SECONDARY;
        break;
      case className.includes(ItemButtonMode.WARNING):
        buttonColor = ThemeMode.WARNING;
        break;
    }
  }

  return (
    <Button
      variant="outlined"
      className={`item-button ${className}`}
      color={buttonColor}
      disabled={isDisabled}
      onClick={onClick}
    >
      {textValue}
      <span className="item-button__inner"></span>
    </Button>
  );
};
