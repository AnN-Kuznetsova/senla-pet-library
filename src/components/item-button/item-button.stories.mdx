import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { ItemButton } from "./item-button";


<Meta title="MDX/ItemButton" component={ItemButton} />

# ItemButton

`ItemButton` - small context button:

<Canvas>
  <ItemButton
    onClick={()=>{}}
    textValue="Ok"
  />
</Canvas>

`ItemButton` - disabled:

<Canvas>
  <ItemButton
    textValue="Ok"
    isDisabled={true}
  />
</Canvas>

`ItemButton` - "more":

<Canvas>
  <ItemButton
    className="item-button--more"
    onClick={()=>{}}
  />
</Canvas>

`ItemButton` - "delete":

<Canvas>
  <ItemButton
    className="item-button--delete"
    onClick={()=>{}}
  />
</Canvas>

`ItemButton` - "close":

<Canvas>
  <ItemButton
    className="item-button--close"
    onClick={()=>{}}
  />
</Canvas>
