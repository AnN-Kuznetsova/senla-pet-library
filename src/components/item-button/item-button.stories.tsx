import * as React from "react";
import styled from "@emotion/styled";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { action } from "@storybook/addon-actions";

import { ItemButton } from "./item-button";


export default {
  title: 'ItemButton',
  component: ItemButton,
  argTypes: {
    //backgroundColor: { control: 'color' },
  },
} as ComponentMeta<typeof ItemButton>;


const Template: ComponentStory<typeof ItemButton> = (args) => <ItemButton {...args} />;


export const Default = Template.bind({});
Default.args = {
  textValue: `Ok`,
  onClick: action(`ItemButton clicked`),
};

export const Disabled = Template.bind({});
Disabled.args = {
  textValue: `Ok`,
  isDisabled: true,
};

export const More = Template.bind({});
More.args = {
  className: `item-button--more`,
  onClick: action(`ItemButton--more clicked`),
};

export const Delete = Template.bind({});
Delete.args = {
  className: `item-button--delete`,
  onClick: action(`ItemButton--delete clicked`),
};


export const Close = Template.bind({});

const CloseButtonWrapper = styled.div`
  position: relative;
  top: -25px;
  width: 100px;
  height: 40px;
`;

Close.args = {
  className: `item-button--close`,
  onClick: action(`ItemButton--close clicked`),
};

Close.decorators = [
  (Story: ComponentStory<React.FC>) => (
    <CloseButtonWrapper>
      <Story />
    </CloseButtonWrapper>
  ),
];
