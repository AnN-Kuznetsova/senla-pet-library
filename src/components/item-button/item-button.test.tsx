import * as React from "react";
import {shallow} from "enzyme";

import {ItemButton, ItemButtonMode} from "./item-button";


describe(`<ItemButton>`, () => {
  it(`Should render button with correct text and class`, () => {
    const itemButtonElement = shallow(
      <ItemButton
        textValue={`Item-Button`}
        className={ItemButtonMode.WARNING}
        onClick={()=>{/**/}}
      />);

    expect(itemButtonElement.length).toBe(1);
    expect(itemButtonElement.text()).toBe(`Item-Button`);
    expect(itemButtonElement.hasClass(ItemButtonMode.WARNING));

    expect(itemButtonElement).toMatchSnapshot();
  });

  it(`Should ItemButton be pressed`, () => {
    const onClick = jest.fn();

    const itemButtonElement = shallow(
      <ItemButton
        onClick={onClick}
      />);

    expect(itemButtonElement.length).toBe(1);
    expect(onClick).toBeCalledTimes(0);

    itemButtonElement.simulate(`click`);
    expect(onClick).toBeCalledTimes(1);

    itemButtonElement.simulate(`click`);
    itemButtonElement.simulate(`click`);
    expect(onClick).toBeCalledTimes(3);
  });
});
