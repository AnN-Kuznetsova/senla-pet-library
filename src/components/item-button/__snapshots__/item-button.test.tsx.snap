// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemButton> Should render button with correct text and class 1`] = `ShallowWrapper {}`;
